### PR TITLE
add missing features for bintray:

### PR DESCRIPTION
**Pull request by arcolife**
- repo creation/deletion
    - repo versioned metadata calculation
    - yaml tags in structs for unmarshalling from any configs

### CODE REVIEW

**Comment by eyalbe4** [comment]
Thank you for this pull request @arcolife!
Please check out the **Guidelines** section in the README.

**Comment by arcolife** [comment]
Sure, let me get back to this.

File: bintray/services/upload.go [hunk 1]
```diff
@@ -197,6 +199,7 @@ func uploadFile(artifact clientutils.Artifact, url, logMsgPrefix string, bintray
 		return false, err
 	}
 	log.Debug(logMsgPrefix+"Bintray response:", resp.Status)
+	fmt.Println(resp.Status, artifact)
```
Comment (eyalbe4) [inline bintray/services/upload.go]: We should probably remove this.

File: bintray/manager.go [hunk 2]
```diff
@@ -117,6 +121,24 @@ func (sm *ServicesManager) IsRepoExists(path *repositories.Path) (bool, error) {
 	return repositoryService.IsRepoExists(path)
 }
 
+func (sm *ServicesManager) CreateReposIfNeeded(path *repositories.Path, params *repositories.Config, repoConfig string) (bool, error) {
```
Comment (eyalbe4) [inline bintray/manager.go]: I suggest renaming this API to *CreateRepoIfNotExist*.

File: bintray/manager.go [hunk 3]
```diff
@@ -117,6 +121,24 @@ func (sm *ServicesManager) IsRepoExists(path *repositories.Path) (bool, error) {
...
+	return repositoryService.CreateReposIfNeeded(path, params, repoConfig)
+}
+
+func (sm *ServicesManager) ExecCreateRepoRest(path *repositories.Path, params *repositories.Config, repoConfig string) (bool, error) {
+	repositoryService := repositories.NewService(sm.client)
+	repositoryService.BintrayDetails = sm.config.GetBintrayDetails()
+	return repositoryService.ExecCreateRepoRest(path, params, repoConfig)
+}
+
+func (sm *ServicesManager) ExecDeleteRepoRest(path *repositories.Path) error {
```
Comment (eyalbe4) [inline bintray/manager.go]: Let's rename to *DeleteRepo*.

**Comment by eyalbe4** [comment]
@arcolife,
Are you planning to continue working in this?

**Comment by arcolife** [comment]
Hi, yes yes. Was delayed due to priority. Is this blocking something? I can do it sometime near end of the upcoming week.

**Comment by eyalbe4** [comment]
No worries @arcolife. It is not blocking anything.
Just wanted to check if this is still relevant.
Thanks

**Comment by eyalbe4** [comment]
Hey @arcolife - 
With Bintray's sunset happening, I assume this PR can be closed, right?

**Comment by arcolife** [comment]
@eyalbe4 sounds right. Thank you

**Pull request closed without merging**